snapcraft/internal/errors.py

```diff
@@ -190,41 +204,46 @@ def __init__(self, filename, message):
 class SnapcraftPartMissingError(SnapcraftError):
 
     fmt = (
-        'Cannot find the definition for part {part_name!r}.\n'
-        'It may be a remote part, run `snapcraft update` '
-        'to refresh the remote parts cache.'
+        'Failed to get part information: '
+        'Cannot find the definition for part {part_name!r}. '
+        'If it is a remote part, run `snapcraft update` '
+        'to refresh the remote parts cache. '
+        'If it is a local part, make sure that it is defined in the '
+        '`snapcraft.yaml`.'
     )
 
 
 class PartNotInCacheError(SnapcraftError):
 
     fmt = (
-        'Cannot find the part name {part_name!r} in the cache. Please '
-        'run `snapcraft update` and try again.\nIf it is indeed missing, '
-        'consider going to https://wiki.ubuntu.com/snapcraft/parts '
-        'to add it.'
+        'Failed to get remote part information: '
+        'Cannot find the part name {part_name!r} in the cache. '
+        'If it is an existing remote part, run `snapcraft update` and try '
+        'again. If it has not been defined, consider going to '
+        'https://wiki.ubuntu.com/snapcraft/parts to add it.'
     )
 
 
 class PluginError(SnapcraftError):
 
-    fmt = 'Issue while loading part: {message}'
+    fmt = (
+        'Failed to load plugin: '
+        '{message}'
+        # FIXME include how to fix each of the possible plugin errors.
+        # https://bugs.launchpad.net/snapcraft/+bug/1727484
+        # --elopio - 2017-10-25
+    )
 
     def __init__(self, message):
         super().__init__(message=message)
 
 
-class PluginNotDefinedError(SnapcraftError):
-
-    fmt = ("Issues while validating snapcraft.yaml: the 'plugin' keyword is "
-           "missing for the {part_name} part.")
-
-
 class SnapcraftPartConflictError(SnapcraftError):
 
     fmt = (
-        'Parts {other_part_name!r} and {part_name!r} have the following file '
-        'paths in common which have different contents:\n'
+        'Failed to run stage step: '
```

Comment:
I’m not sure that adding “Failed to run stage step:” is an improvement. Should I know what a “stage step” is?